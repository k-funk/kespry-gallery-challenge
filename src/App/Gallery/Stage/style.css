.stage {
  position: relative;
}

.stage .select-image {
  align-items: center;
  background-color: rgba(255, 255, 255, 0);
  color: #fff;
  cursor: pointer;
  display: flex;
  font-size: 40px;
  height: 100%;
  justify-content: center;
  position: absolute;
  top: 0;
  transition: all var(--transition-time);
  width: 75px;
}

.stage .select-image:hover {
  background-color: rgba(255, 255, 255, .3);
}

.stage .select-image.previous {
  left: 0;
}

.stage .select-image.next {
  right: 0;
}

.stage img {
  margin: 0 auto;
  width: 100%;
}

.stage img::selection {
  background: transparent;
}
