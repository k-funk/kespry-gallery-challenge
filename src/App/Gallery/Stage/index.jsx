import React from 'react';
import { PropTypes as T } from 'prop-types';

import './style.css';


const Stage = ({ img, selectNextImage, selectPreviousImage }) => {
  if (!img) { return null; }

  return (
    <div className="stage">
      <div
        className="select-image previous"
        role="button"
        onClick={selectPreviousImage}
        onKeyUp={selectPreviousImage}
        tabIndex={0}
      >
        ❮
      </div>
      <div
        className="select-image next"
        role="button"
        onClick={selectNextImage}
        onKeyUp={selectNextImage}
        tabIndex={0}
      >
        ❯
      </div>
      <img
        src={img}
        alt={img}
      />
    </div>
  );
};

Stage.propTypes = {
  img: T.string,
  selectNextImage: T.func.isRequired,
  selectPreviousImage: T.func.isRequired,
};

export default Stage;
