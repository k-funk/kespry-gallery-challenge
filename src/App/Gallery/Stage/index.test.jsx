import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import Stage from './index';


describe('outputs the expected tree when', () => {
  let wrapper;
  let selectNextImageSpy;
  let selectPreviousImageSpy;
  let defaultProps;

  beforeEach(() => {
    selectNextImageSpy = jest.fn();
    selectPreviousImageSpy = jest.fn();
    defaultProps = {
      selectNextImage: selectNextImageSpy,
      selectPreviousImage: selectPreviousImageSpy,
    };
  });

  test('with an img', () => {
    wrapper = shallow((
      <Stage {...defaultProps} img="https://img.com/1" />
    ));
  });

  test('no img', () => {
    wrapper = shallow((
      <Stage {...defaultProps} img={null} />
    ));
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
