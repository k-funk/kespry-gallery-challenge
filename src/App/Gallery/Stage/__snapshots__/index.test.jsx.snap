// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when no img 1`] = `""`;

exports[`outputs the expected tree when with an img 1`] = `
<div
  className="stage"
>
  <div
    className="select-image previous"
    onClick={[MockFunction]}
    onKeyUp={[MockFunction]}
    role="button"
    tabIndex={0}
  >
    ❮
  </div>
  <div
    className="select-image next"
    onClick={[MockFunction]}
    onKeyUp={[MockFunction]}
    role="button"
    tabIndex={0}
  >
    ❯
  </div>
  <img
    alt="https://img.com/1"
    src="https://img.com/1"
  />
</div>
`;
